src/treetensornetworks/solvers/dmrg.jl

```diff
@@ -1,58 +1,34 @@
-function eigsolve_solver(;
-  solver_which_eigenvalue=:SR,
-  ishermitian=true,
-  solver_tol=1e-14,
-  solver_krylovdim=3,
-  solver_maxiter=1,
-  solver_verbosity=0,
-)
-  function solver(H, init; normalize=nothing, region=nothing, half_sweep=nothing)
-    howmany = 1
-    which = solver_which_eigenvalue
-    vals, vecs, info = eigsolve(
-      H,
-      init,
-      howmany,
-      which;
-      ishermitian,
-      tol=solver_tol,
-      krylovdim=solver_krylovdim,
-      maxiter=solver_maxiter,
-      verbosity=solver_verbosity,
-    )
-    psi = vecs[1]
-    return psi, (; solver_info=info, energies=vals)
-  end
-  return solver
-end
-
 """
 Overload of `ITensors.dmrg`.
 """
+
+function dmrg_sweep(
+  nsite::Int, graph::AbstractGraph; root_vertex=default_root_vertex(graph)
+)
+  return tdvp_sweep(2, nsite, Inf, graph; root_vertex, reverse_step=false)
```

Comment:
```suggestion
  order = 2
  time_step = Inf
  return tdvp_sweep(order, nsite, time_step, graph; root_vertex, reverse_step=false)
```
so we remember what `2` and `Inf` mean.

Reply:
Also, I'm not sure how I feel about calling this `dmrg_sweep` since it could be used by other solvers like `linsolve`. I'm also not a huge fan of calling it `tdvp_sweep` for the same reason.

Maybe we can rethink this API a bit, for example rename the current `tdvp_sweep` to `default_sweep_plan` and then just call that with different inputs from within the different solvers. I think a good design for that would be to choose default values that "just work" for DMRG, including defaulting to `order=2`, and then provide optional inputs like `reverse_step=true` and `time_step` for use with TDVP. That may mean moving some of the inputs like order and time step to keyword arguments, which seems like a better design anyway.

Reply:
Yes, and this is what the `default_sweep_regions` function was trying to be. That is, a sort of "typical" sweep plan that most algorithms (dmrg, linsolve, etc.) could use. We could definitely think about moving that function out of the `update_step.jl` file but the intent there was for it to be a pretty generic default.

Reply:
I see, I forgot about `default_sweep_regions`, I'm not sure what the fate of that is in the current PR since a lot of things are getting changed in this PR. But it seems like `default_sweep_regions`, `tdvp_sweep`, and `dmrg_sweep` should all get consolidated into a single function, which tentatively we are thinking of calling something like `default_sweep_plan`, `default_region_updates`, or `default_region_update_plans` (since it contains both the regions that will be updated but also information about how to do the update), which I think should just get called directly by the different solvers like `dmrg`, `tdvp`, `linsolve`, etc. and passed to `alternating_update`.

Opinions on what to name that function would be welcome. We want to come up with a good name for that function and what it outputs which we can then use consistently throughout the rest of the code. The list output by `tdvp_sweep`/`dmrg_sweep` is called `region_updates` in the current version of this PR but we are discussing alternative names, which hopefully will be consistent with the name of the new function.

Reply:
I incorporated the suggestion, and the sweep_regions constructor is planend to be refactored in an upcoming PR